lisp/magit-base.el

```
@@ -1215,6 +1226,18 @@ Like `message', except that `message-log-max' is bound to nil."
          (goto-char (or ,pos 1))
          ,@body))))
 
+(defun magit--ellipsis (&optional where)
+  "Build an ellipsis always as string, depending on WHERE."
+  (if-let ((default (alist-get nil magit-ellipsis)))
+      (pcase-let ((`(,fancy . ,universal) (car (alist-get where magit-ellipsis default))))
```

The "default" key is the most used key, so I find it a bit odd to always look it up twice. I also think that error message below is worded as it is because of how you implemented this here. Simply saying that `magit-ellipsis` is invalid, seems more appropriate.

Speaking of invalid values, I think this should be backward compatible. There's no need to inform users that the format has been changed, and the old format should just be silently accepted.